packages/gatsby/src/utils/app-data.ts

```diff
@@ -0,0 +1,13 @@
+import fs from 'fs-extra'
+import path from 'path'
+
+const APP_DATA_JSON = `app-data.json`
+
+export const write = (publicDir: string, hash: string): Promise<void> => {
+  return fs.outputJson(path.join(publicDir, `page-data`, APP_DATA_JSON), {
```

Comment:
I added a return statement  because it is awaiting in build.js.

https://github.com/gatsbyjs/gatsby/blob/ea4b4c23842140e004c74d1baa9cfc1284fbfa1a/packages/gatsby/src/commands/build.js#L138